### ISSUE TITLE
[vm_deploy] does not check for available disk capacity

### ISSUE THREAD

**Issue by TGM** [OpenNebula/one#6462]
**Description**
You can deploy any number of VMs on the same datastore as long as the disk size is under the TOTAL_MB size.
In other works, vm_deploy does not check the ALLOCATED capacity, and it can't since the ALLOCATED capacity is not a computed parameter.

This is a HUGE issue, due to the fact that when the allocated space starts to fill up the VMs will start crashing due to overprovisioning.

**To Reproduce**
Create a new datastore with 30GB.
Create 2 new VM under this datastore with 20GB each.

**Expected behavior**
The 1st VM should provision.
The 2nd VM should fail due to unavailable space. 

**Details**
The 2nd VM does not fail, as there is indeed free space on the DISK due to _thin_ provisioning. Other types of provisioning are available mostly for vCenter. 
It should fail to provisioning returning an error with not available space.

**Comment by Franco-Sparrow** [comment]
I agree with your @TGM , but this more like a feature request than a bug. I think the idea of the cloud is to give you the chance to create VMs as many as you want, and the cloud admin should have a monitoring system to check the FS usage....if you are running out of space, you know you need to add more disks to your cloud, so user dont be affected.

**Comment by kurojishi** [comment]
while the idea is that, without a metric for allocation, it's impossible to know overindexing of resources before it's too late.

It's not impossible to have a case where lots of vm gets allocated and suddenly all the guests start generating data that fills a lot of storage, causing failures all over the place if overallocated over certain amounts.

So one should be able to overallocate, but one should be very much aware of how much over-allocation is happening, so this information is paramount to have for proper management of a large platform, both in metric form and something that's easy to see from dashboards
